Assets/GameEngine.cs

```diff
@@ -6,17 +6,35 @@ public class GameEngine : MonoBehaviour {
     private Transform source;
 
 	public void RegisterStrongholdClick(Transform selected) {
-        if (source == null) {
+
+        ObjectOwnerState selectedInfo = selected.GetComponent<ObjectOwnerState>();
+
+        if (source == null && selectedInfo.player!=null) {
             // If source is not yet selected, selected stronghold is the source
             source = selected;
-        } else {
+
+            //Turn Source StrongHold Green
+            selectedInfo.rend.material.shader = Shader.Find("Outlined/UltimateOutline");
+            selectedInfo.rend.material.SetColor("_FirstOutlineColor", Color.green);
+            selectedInfo.rend.material.SetColor("_SecondOutlineColor", Color.green);
+
+        } else if(source){
```

Comment:
Nitpick: Just to be consistent, it'll be good to have spaces after `if` and before the `{`.